cpp/src/arrow/util/logging.h

```diff
@@ -186,6 +189,8 @@ class ARROW_EXPORT ArrowLog : public ArrowLogBase {
   // In InitGoogleLogging, it simply keeps the pointer.
   // We need to make sure the app name passed to InitGoogleLogging exist.
   static std::unique_ptr<std::string> app_name_;
+  // Keep the log directory.
```

Comment:
Do both variables need to be kept publicly? They could be private static in `InstallFailureSignalHandler`.

Reply:
Will change it for `app_name_`, but for `log_dir_`, this variable will be used in 2 functions. It is better to keep it as a static variable.

Reply:
but it doesn't need to be public, you can move it into the implementation (logging.cc).

Reply:
Will do.

Reply:
@fsaintjacques  Thanks for the comment!

Reply:
@fsaintjacques Do you have more comments on this PR？

Reply:
good for me.